apps/calendar/js/app.js

```diff
@@ -23,225 +24,185 @@ var storeFactory = require('store/factory');
 var timeObserver = require('time_observer');
 var viewFactory = require('views/factory');
 
-var pendingClass = 'pending-operation';
+var loadLazyStyles = null;
+var pendingManager = new PendingManager();
+var startingURL = window.location.href;
+
+function setupCore(dbName) {
+  if (core.db) {
+    return;
+  }
+  core.db = new Db(dbName || 'b2g-calendar');
+  core.errorController = new ErrorController();
+  core.notificationsController = notificationsController;
+  core.periodicSyncController = periodicSyncController;
+  core.providerFactory = providerFactory;
+  core.serviceController = new ServiceController();
+  core.storeFactory = storeFactory;
+  core.syncController = new SyncController();
+  core.timeController = new TimeController();
+  core.viewFactory = viewFactory;
+}
+
+function setupRouter() {
```

Comment:
I moved all this logic into the `router.js` file while I was working on `hashchange` (https://github.com/mozilla-b2g/gaia/pull/28873) and `app.js` only called `router.start()`, I think it makes more sense since we are not really changing the routes dynamically - reduces the noise inside `app.js`

not really important/priority but since it changed whole app.js it might make sense to do it now as well.